File: Javascript/backend_nest/src/infra/db/inmemory/mappers/UserInMemoryMapper.ts
import { User } from 'src/domain/entities/User';
import { Mapper } from 'src/domain/kernel/Mapper';
import { UserInMemory } from '../entities/UserInMemory';

export class UserInMemoryMapper extends Mapper<User, UserInMemory> {
  toDomain(external: UserInMemory): User {
    return User.create({
      id: external.id,
      email: external.email,
      role: external.role,
      firstName: external.firstName,
      lastName: external.lastName,
    });
  }

  toExternal(domain: User): UserInMemory {
    return new UserInMemory(domain.id, domain.email, domain.role, domain.firstName, domain.lastName);
  }
}

File: Python/backend_flask/infra/db/inmemory/mappers/user_inmemory_mapper.py
from domain.entities.User import User
from domain.kernel.Mapper import BaseMapper
from infra.db.inmemory.entities.user_inmemory import UserInMemory

class UserInMemoryMapper(BaseMapper[User, UserInMemory]):
    def to_domain(self, user_in_memory: UserInMemory) -> User:
        return User.create(
            id=user_in_memory.id,
            email=user_in_memory.email,
            role=user_in_memory.role,
            first_name=user_in_memory.first_name,
            last_name=user_in_memory.last_name
        )

    def to_external(self, user: User) -> UserInMemory:
        return UserInMemory(
            id=user.id,
            email=user.email,
            role=user.role,
            first_name=user.first_name,
            last_name=user.last_name
        )